torch/profiler/_pattern_matcher.py

```diff
@@ -33,8 +40,10 @@ def match(self, event: _ProfilerEvent):
         raise NotImplementedError
 
     def matched_events(self):
+        if self.skip:
+            return []
         matched_events = []
-        for event in eventTreeDFS(self.event_tree):
+        for event in eventTreeBFS(self.event_tree):
```

Comment:
Its because I need to have access to the events in the same level in the order as they take place. But you could also argue that I could just reverse the order that I add children into stack. What do you think makes more sense?

Reply:
I thought the semantics of `match` were that you could traverse the events in any order?

Reply:
Maybe this is a sign that match should be matching a higher level node?

Reply:
Uh yes, but the time complexity would be too high if i don't keep a visited set and avoid. redundant visited. For example, one aten::select node in a for loop implies that all the aten::select in the for could be a match. By keeping a visited set I can avoid revisited these. But that requires me to traverse in the particular order..

Reply:
In that case it should probably be part of the API so the matcher can specify the traversal order that it expects.